internal/controller/manifests_downloader.go

```diff
@@ -75,15 +76,25 @@ func (p *phaseReconciler) downloadManifests(ctx context.Context) (reconcile.Resu
 		return reconcile.Result{}, wrapPhaseError(err, operatorv1.ComponentsFetchErrorReason)
 	}
 
+	spec := p.provider.GetSpec()
+
+	if spec.Version == "" {
+		// User didn't set the version, try to get repository default.
+		spec.Version = repo.DefaultVersion()
+
+		// Add version to the provider spec.
+		p.provider.SetSpec(spec)
```

Comment:
The reason spec is set twice is because it accommodates two distinct workflows:

- When downloading manifests from GitHub/GitLab.
- When working with config maps containing manifests.

For the first workflow, the version is set during the manifest download process when a fetch URL is used. In the case of config maps, the version setting occurs in the load phase. Regardless of the workflow, the spec update is executed only once.